src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -206,15 +215,33 @@ private boolean areAllChildrenIgnored() {
     }
 
     /**
-     * Returns a {@link Statement}: run all non-overridden {@code @BeforeClass} methods on this class
+     * Returns a {@link Statement}: run all non-overridden {@code @Ignore} methods on this class
+     * and superclasses before executing {@code next}.
+     */
+    protected Statement withIgnores(final RunNotifier notifier) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                for (final T each : getFilteredChildren()) {
```

Comment:
Yes, each doesn't need to be final. This is a safety programming approach, which prevents assigning each with a different content. If this is against your coding style, I will remove it.